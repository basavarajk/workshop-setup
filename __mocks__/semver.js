module.exports = {
  satisfies: jest.fn(),
}
