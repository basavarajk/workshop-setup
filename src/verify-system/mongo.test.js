import cpMock from 'child_process'
import {satisfies as satisfiesMock} from 'semver'
import validateMongo from './mongo'

jest.mock('child_process')

test('returns null if conditions are satisfied', () => {
  setup()
  const desired = '3.4.5'
  const result = validateMongo(desired)()
  expect(result).toBe(null)
})

test('returns version mismatch error', () => {
  const mockActual = '3.8.2'
  const mockExecResult = `db version blah ${mockActual} blah`
  setup({satisfies: false, mockExecResult})
  const desired = '^2.0.0'
  const result = validateMongo(desired)()
  expect(result).toContain(desired)
  expect(result).toContain(mockActual)
  expect(result).toMatchSnapshot()
})

test('returns mongod command not found error', () => {
  cpMock.execSync.mockClear()
  cpMock.execSync.mockImplementation(() => {
    throw new Error('blah')
  })
  const desired = '^2.0.0'
  const result = validateMongo(desired)()
  expect(result).toMatch(/mongod.*isn't available.*PATH/)
  expect(result).toMatchSnapshot()
})

function setup(
  {
    mockExecResult = 'db version blah blah 3.4.5',
    satisfies = true,
  } = {},
) {
  satisfiesMock.mockClear()
  cpMock.execSync.mockClear()
  satisfiesMock.mockImplementation(() => satisfies)
  cpMock.execSync.mockImplementation(() => mockExecResult)
}
