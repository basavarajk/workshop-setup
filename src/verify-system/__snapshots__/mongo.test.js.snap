// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns mongod command not found error 1`] = `"There was an error determining your mongo version. This could be because the \`mongod\` command isn't available in your PATH. Learn how to fix this on windows (https://www.howtogeek.com/118594/how-to-edit-your-system-path-for-easy-command-line-access/) or on mac (http://stackoverflow.com/a/24322978/971592)."`;

exports[`returns version mismatch error 1`] = `"Your version of mongo (3.8.2) does not satisfy the desired range of ^2.0.0. Please install a more recent version: https://www.mongodb.com/download-center"`;
